libraries/chain/controller.cpp

```diff
@@ -4701,6 +4702,25 @@ struct controller_impl {
       return conf.block_validation_mode == validation_mode::LIGHT || conf.trusted_producers.count(producer);
    }
 
+   bool should_terminate() const {
+      constexpr block_num_type max_reversible_blocks = 1000;
```

Comment:
Temporary solution until we can get the consensus level change that will restrict producing higher block heights until finalizers come back online.

Reply:
We have some libtester tests that failed because they produce more than 1000 blocks (more than 2300 if I counted correctly). How about we go with 30 minutes; 3600 blocks.

Reply:
As long as we have some protection against OOM, 30 minutes is fine for now.